### ISSUE TITLE
[Neo.DatabaseError.General.UnknownError]: Tried to invalidate a cached property CachedProperty

### ISSUE THREAD

**Issue by ofraiwan** [neo4j/neo4j#12508]
A strange exception was thrown by neo4j 4, the same query is being used on neo4j 3.5 without any unexpected behavior.

The exact exception is:
`ERROR Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: Tried to invalidate a cached property CachedProperty(a,Variable(a),PropertyKeyName(uploaded),NODE_TYPE) but no slot was found for the entity name in SlotConfiguration(longs=3, refs=1, slots=Map(  p@7 -> LongSlot(0,false,Node), space -> LongSlot(2,false,Node),   p_rel@31 -> LongSlot(1,false,Relationship)), cachedProperties=Map(CachedProperty(a,Variable(a),PropertyKeyName(uploaded),NODE_TYPE) -> RefSlot(0,false,Any)))., reference ca2956eb-df91-4359-9bb6-9330109dbe11.`

The query is as follows
```
MATCH (p:Post {id: $post_id})-[p_rel:IN_PART]->(space)
REMOVE p._lock
WITH p, p_rel
OPTIONAL MATCH (a:Attachment)-[:ATTACHED_TO]->(p)
WHERE a.has_file AND (NOT a.uploaded OR (a.has_thumb AND NOT a.thumb_uploaded))
WITH p, p_rel, count(a) as missing_count
WHERE missing_count = 0
SET p.completed = true
WITH p, p_rel
CALL apoc.refactor.setType(p_rel, 'IN') yield input, output
RETURN p
```
The query basically set a property on nodes of type **Post**, if certain conditions are met on its child nodes of type **Attachment**.

**Setup:**
Neo4j Version: 4.0.3
Operating System: Ubuntu 18.04.3 LTS
DB Mode: Causal Cluster

Hint: removing the p._lock line will not raise the exception.

However, the query was running successfully without any exceptions on neo4j 3.5.

**Comment by sherfert** [comment]
Thanks for the bug report. We'll look into it.

**Comment by mnd999** [comment]
This issue should be resolved from Neo4j 4.0.5 onwards.
